### ISSUE TITLE
Resilvering restarts after each pool import

### ISSUE THREAD

**Issue by bimonsubio1984** [openzfs/zfs#8845]
Hello,

I have found this problem in my old pool after replicating to its different mirrors, say there is a mirror B_1 and B_2 for a pool Backup.

1) I have offlined device B_2 and replicated my info to pool on device B_1.

2) I have offlined device B_1 and onlined device B_2 and replicated more recent copy of my info to pool on device B_1.

3) I onlined both devices and resilvering passed. But after exporting the pool resilvering started again. I tried to import from a single device and each time it begins to resilver on a single device.

4) I have copied all datasets from B_1 to another pool created on B_2. Then I attached B_1 as a mirror and it still continues to resilver indefinitely amount of times even after passing 100% of resilvering the first time. Actually first time it failed to resilver due to some problems with disks described in the next item

5) It seems my browser Brave has some type of a trojan which manages devices via kernel though it runs under a non privileged user in a non sandbox mode without sid bit.
I have experienced HDD clicking sounds and /dev/sdX devices for this pool lost during using Brave browser and posting some unwanted by proprietary parties information on forums.
Never it clicked when Brave was not started, so I conclude it may be a reason of the problem with disks lost sometimes. All 4 disks are lost, powering them off and on helps them to reappear.
I have experienced this problem with Brave many times, may be 5 or more times.

How can I fix this to stop resilvering?

modinfo zfs | grep -iw version; modinfo spl | grep -iw version 
version:        0.7.12-1~bpo9+1
version:        0.7.12-1~bpo9+1
17:23 root@workstation > uname -a
Linux workstation 4.19.36-gnu #1.0 SMP Tue Sep 27 12:35:59 EST 1983 x86_64 GNU/Linux

cat /etc/devuan_version 
ascii

**Comment by bimonsubio1984** [comment]
It seems your recent ZFS versions cannot be used even in home environment.

v0.6.x was more reliable for me even in production.

**Comment by bimonsubio1984** [comment]
I would suggest anyone to use Illumos like SmartOS for free ZFS until this issue is fixed.

**Comment by bimonsubio1984** [comment]
Btw, are pools not binary compatible with Illumos anymore?

I was not able to import my ZOL v0.7.12 pool from OpenIndiana.

Is is still possible at least to send/receive them to OpenIndiana via a channel like SSH?

**Comment by johnnyjacq16** [comment]
@bimonsubio1984  could you try the following:
`zpool set cachefile=/etc/zfs/zpool.cache <name of pool>`
`systemctl restart zfs-import-cache` (restart the zfs-import-cache service on OpenIndiana).

That being said the resilvering doesn't seem to be a issue, zfs will deal with the resilvering on this own no manual intervention required. So if two devices are in a mirror, once you remove one of the device in the mirror and place the device back into the mirror a resilvering should take place to  sync everything. What issues are you having for the new version of zfs?

**Comment by bimonsubio1984** [comment]
I tried to import pool with only one part of the mirror each time
It was resilvering for a very long time each part like:

![ZOL6](https://user-images.githubusercontent.com/44701154/58774059-311ea480-85d9-11e9-9efc-e822ccbcf80f.jpg)

And then after importing the whole pool it resilvered again for a short time and 
![ZOL7](https://user-images.githubusercontent.com/44701154/58774111-5d3a2580-85d9-11e9-8e42-d28cad87d19b.jpg)

**Comment by bimonsubio1984** [comment]
Now it does not resilver anymore finally without a good reason.

I still experience disks lost even without browser, so I conclude there is at least a backdoor in a new 4 port Marvel controller from AliExpress. Most likely  this backdoor is managed by a remote radio channel.

It was working fine during about a couple of months and recently started to lose all disks at once when posting to forums where I get banned and also at important times when was trying to combine an earlier spited mirrors of the pool.

**Comment by johnnyjacq16** [comment]
Single vdev without its mirror is being resilvered means that zfs detected that there are disks missing in the pool, when you remove or offline the device at which point the pool is in a degraded state, zfs will allow you to replace the disk by getting a new disk and while both new and old disk are connected you can replace the disk or in your case where zfs thinks that you removed the old disk and are about to connect the new disk for replacement (the new disk will automatically get all the data that was of the old disk even though you have already remove the old device).

Single disk have no redundancy and if copies=1 on all datasets in your pool, there is only one copy of the data for the dataset. `zfs list` will show you all your dataset.
zfs prefer to work directly with the disk and working through a raid controller is less preferred.

**Comment by bimonsubio1984** [comment]
Can you please suggest, can Adaptec 3805 work with ZFS in JBOD mode?
Can it support 3TB disks?

**Comment by bimonsubio1984** [comment]
Earlier I did not see resilivering started after offlining (powered off it by a button) a mirrored vdev.
I powered off the HDDs BEFORE importing, so the pool was imported only with a single part of mirror.

I keep a couple of HDDs generally off to avoid a damage by power surge.
Later when I onlined them, they were resilvered in a few minutes always.

But several last times I got continuous resilvering restarted each time I imported the pool, does not matter all vdevs, or a single mirror.

**Comment by johnnyjacq16** [comment]
With the use of a raid controller, if there is a special need for it then by all means continue to use it whether it be a hardware or/and software raid controller, that said i'd say that a hardware would be better than a software raid controller. 
However using no raid controller with zfs is the best use case scenario with zfs as zfs already has raid capabilities built into it. 

zfs will generally resilver if it detects any changes, so exporting then importing the pool may cause a resilver to happen, it zfs way to check to ensure that all vdevs in a mirror configuration is synced up.

**Comment by bimonsubio1984** [comment]
Generally a small light resilvering takes about 10 minutes after import.
Even full pool sync takes about 14 hours.
And you saw long resilvering took 3 times more.

**Comment by Am1GO** [comment]
Reading such issue reports makes me think that the code of conduct should be heavily relaxed.

**Comment by bimonsubio1984** [comment]
What exactly do you find as trolling. In my opinion you are trolling.

**Comment by kpande** [comment]
sounds like defective hw, not a trojan or backdoored hw.

**Comment by bimonsubio1984** [comment]
Hardware was good for years and after posting some messages to:
http://dev1galaxy.org/viewtopic.php?id=2818
it became defective instantly especially during posting almost each time, fantastic.

**Comment by bimonsubio1984** [comment]
It failed again during posting on forums related to conspiracy. 

Replaced with two old controllers, each part of mirror has its own obsolete controller now and all works exceptionally. Concluding there were less backdoors in hardware produced 10 years ago.

**Comment by bimonsubio1984** [comment]
New controller which I had to replace was based on Marvel 88SE9215:

https://ru.aliexpress.com/item/32921298195.html

It has many positive opinions of over 100 happy buyers. 
It worked fine for me until posting on specific forums too.

**Comment by bimonsubio1984** [comment]
I have found a bug most likely. 
When importing the whole pool resilvering restarts from 0% at about 10% and it happens many times.

So I imported the pool with only a single mirror part, now it imports as alive without IO suspended as earlier, I guess metadata was corrupted earlier and now it is restored.

**Comment by bimonsubio1984** [comment]
Another part of pool is still havily damaged:

![Part2](https://user-images.githubusercontent.com/44701154/59338093-2b346c00-8d1b-11e9-9616-ff966642bec7.png)

**Comment by bimonsubio1984** [comment]
Think twice before using backdoored hardware from China

**Comment by bimonsubio1984** [comment]
Another part of pool is still heavily damaged:

![Part2B](https://user-images.githubusercontent.com/44701154/59338393-b6156680-8d1b-11e9-85ae-c0423c048856.png)

**Comment by bimonsubio1984** [comment]
I imported alive part of the pool with B1 and B3, it resilvered successfully.
Then reimported the pool with B1+B2 and B3, it resilvered successfully too.
After importing full pool with B1+B2 and B3+B4 resilvering of B_4 restarts at about reaching 10% many times even WITHOUT exporting and importing again as it was earlier.

So detached B4 and reattached it again into the mirror. 

Compared original large set of files about 2TB using rsync --dry-run and found Backup is completely consistent. So ZFS was able to recover B1+B2 and B3, but was not able to recover B4 which I had to reattach again to sync it from scratch.
